### PR TITLE
[5.5] Allows touch method to accept other column

### PR DESCRIPTION
**Pull request by mul14**
So we can use something like

```php
Post::find(1)->touch('published_at');

User::find(1)->touch('activated_at');
```

### CODE REVIEW

**Comment by lucasmichot** [comment]
@mul14 , so this means that after `Post::find(1)->touch('published_at');`, `updated_at` will also be updated, right ?

---

Tests?

File: src/Illuminate/Database/Eloquent/Concerns/HasTimestamps.php [hunk 1]
```diff
@@ -16,28 +16,34 @@
     /**
      * Update the model's update timestamp.
      *
+     * @param  mixed  $column
```
Comment (lucasmichot) [inline src/Illuminate/Database/Eloquent/Concerns/HasTimestamps.php]: `@param  string|null  $column`

File: src/Illuminate/Database/Eloquent/Concerns/HasTimestamps.php [hunk 2]
```diff
@@ -16,28 +16,34 @@
...
-        $this->updateTimestamps();
+        $this->updateTimestamps($column);
 
         return $this->save();
     }
 
     /**
      * Update the creation and update timestamps.
      *
+     * @param  mixed  $column
```
Comment (lucasmichot) [inline src/Illuminate/Database/Eloquent/Concerns/HasTimestamps.php]: `@param  string|null  $column`

**Comment by mul14** [comment]
@lucasmichot Yes, you are right.

File: src/Illuminate/Database/Eloquent/Concerns/HasTimestamps.php [hunk 3]
```diff
@@ -16,28 +16,34 @@
...
      * @return void
      */
-    protected function updateTimestamps()
+    protected function updateTimestamps($column = null)
     {
         $time = $this->freshTimestamp();
 
+        if ($column) {
```
Comment (lucasmichot) [inline src/Illuminate/Database/Eloquent/Concerns/HasTimestamps.php]: You should probably also check that `$colum` is a date:
`if ($column && $this->isDateAttribute($column)) {`

Comment (mul14) [inline src/Illuminate/Database/Eloquent/Concerns/HasTimestamps.php]: Thanks for review. 

Is the `isDateAttribute` method already exists? I can't find it.

Comment (lucasmichot) [inline src/Illuminate/Database/Eloquent/Concerns/HasTimestamps.php]: It does in `\Illuminate\Database\Eloquent\Concerns\HasAttributes::isDateAttribute`
`\Illuminate\Database\Eloquent\Concerns\HasAttributes` is used by `\Illuminate\Database\Eloquent\Model`

Comment (mul14) [inline src/Illuminate/Database/Eloquent/Concerns/HasTimestamps.php]: Thanks, it works after I add `protected $dates` to the model.

**Comment by taylorotwell** [comment]
Just do:

```
Post::find(1)->update(['column' => new DateTime]);
```

**Pull request closed without merging**